### DISCUSSION TITLE
conditional control to switch preconditioner

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#25225] [Q&A General] [answered]
Hi, is there a way to switch between two preconditioner blocks as a function of a variable extreme (minimum) value?
I'm aware of the test for switching between BCs:
```
modules/navier_stokes/test/tests/finite_volume/controls/switch-pressure-bc/test.i
```
but something similar does not seem to apply to the Preconditioning block

**Comment by lindsayad** [comment, marked as answer]
No on-the-fly switching of preconditioner blocks is not currently supported. I don't know if any of the discussion in https://github.com/idaholab/moose/discussions/25063 would be of interest to you

**Reply by garciapintado** [reply]
Thanks!

**Reply by GiudGiud** [reply]
This seems like a feature request.
I raised #25233

**Reply by garciapintado** [reply]
Many thanks @GiudGiud, as we can name blocks for different preconditioner blocks with just a "integer" number labels, would it work and is possible somehow to allow the use of a ParsedFunction [returning an integer, based on the extrema of a variable, taken from a postprocessor] within an fparse expression? I mean something as:

[Preconditioners]
active = ${fparse ${switch_function}}     # where the Function switch_function would return an integer
  [0]
    ...
  []
  [1]
    ...
  []
[]

**Reply by GiudGiud** [reply]
We don’t support that syntax. 
i think there s likely some non trivial petsc calls to make to switch preconditioners, we would probably look at supporting this with the Controls system first. 

if you want to work on adding this we can support you. Otherwise you ll have to leverage the restart system to switch preconditioniers

**Comment by garciapintado** [comment]
Thanks! Was just a curiosity. Support via the Controls system would be just great!
